src/engine/channels/enginedeck.cpp

```diff
@@ -44,6 +49,29 @@ EngineDeck::~EngineDeck() {
     delete m_pPregain;
 }
 
+void EngineDeck::processStem(CSAMPLE* pOut, const int iBufferSize) {
+    int stereoChannelCount = m_pBuffer->getChannelCount() / mixxx::kEngineChannelOutputCount;
+    auto allChannelBufferSize = iBufferSize * stereoChannelCount;
+    if (m_stemBuffer.size() < allChannelBufferSize) {
+        m_stemBuffer = mixxx::SampleBuffer(allChannelBufferSize);
+    }
+    m_pBuffer->process(m_stemBuffer.data(), allChannelBufferSize);
+
+    // TODO(XXX): process effects per stems
+
+    SampleUtil::clear(pOut, iBufferSize);
```

Comment:
This is used to set the buffer to zero, before mixing (summing) all the stereo channel in there. Note that my first version used to copy the first stereo channel into pOut, then the merge would only happens for the next channels. However, I simplified this because of what is in the next PR with stem gain, so I thought I will keep the structure as it needs to be for the future.

Not sure if that make sense, but to summarize, I will need this for loop later, so I thought I will use it, even if it could be simplified at that stage.